### PR TITLE
Module 3 task 1

### PR DESCRIPTION
**Pull request by gecmenow**
I have added folder "V3_KramDeliveryFood" for third task.

### CODE REVIEW

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Buyers.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Buyers
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Buyers.cs:5]: Buyer

Comment (yevhenbilushko) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Buyers.cs:5]: i would name it "client"
up to you

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Categories.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Categories
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Categories.cs:5]: Category
please rename all models to singular

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Delivers.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace KramDeliveryFood_v3.Models
+{
+    public enum Delivers
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Delivers.cs:3]: it is not db model
so better to separete it and move from model folder

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/OrderStatus.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+﻿namespace KramDeliveryFood_v3.Models
+{
+    public enum OrderStatus
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/OrderStatus.cs:3]: it is not db model
so better to separete it and move from model folder

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Orders.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        public List<Products> OrderProducts { get; set; }
+        public Buyers Buyer { get; set; }
+        public string PhoneNumber { get; set; }
+        public string Address { get; set; }
+        public DateTime OrderTime { get; set; }
+        public decimal FullPrice { get; set; }
+        public Delivers Delivers { get; set; }
+        public OrderStatus OrderStatus { get; set; }
+
+        public Orders()
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Orders.cs:17]: better not to have methods 
cause it is db model

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Products.cs [hunk 6]
```diff
@@ -0,0 +1,104 @@
+﻿using System;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Products
+    {
+        public Guid Id { get; set; }
+        private string _name { get; set; }
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Products.cs:8]: please rename all properties in common style
Name
CategoryId
and others

Comment (yevhenbilushko) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Products.cs:8]: and leave only properties with { set get}

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Tariffs.cs [hunk 7]
```diff
@@ -0,0 +1,8 @@
+﻿namespace KramDeliveryFood_v3.Models
+{
+    public enum Tariffs
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Tariffs.cs:3]: move from model folder

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Buyers.cs (hunk 1 above)
Comment (gecmenow) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Buyers.cs]: I left buyer, but thanks for advice

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Data/StoreContext.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
+﻿using KramDeliveryFood_v3.Interfaces;
+using System.Collections.Generic;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class StoreContext : IData
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Data/StoreContext.cs:6]: it is no need for now
only DB structure

Comment (yevhenbilushko) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Data/StoreContext.cs:6]: remove it please

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Interfaces/IData.cs [hunk 9]
```diff
@@ -1,15 +1,13 @@
-﻿using KramDeliverFoodCompleted.Models;
+﻿using KramDeliveryFood_v3.Models;
 using System.Collections.Generic;
 
-namespace KramDeliverFoodCompleted.Interfaces
+namespace KramDeliveryFood_v3.Interfaces
 {
     public interface IData
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Interfaces/IData.cs:6]: no need it 
remove it

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+using System;
+using System.Collections.Generic;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Order
+    {
+        public Guid OrderId { get; set; }
+        public List<Product> OrderProducts = new();
+        public Buyer Buyer { get; set; }
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:11]: add BuyerId

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+        public string PhoneNumber { get; set; }
+        public string Address { get; set; }
+        public DateTime OrderTime { get; set; }
+        public decimal FullPrice { get; set; }
+        public Deliver Delivers { get; set; }
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:16]: Delivers - list
Deliver - one entity

Comment (yevhenbilushko) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:16]: make decision

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs [hunk 12]
```diff
@@ -0,0 +1,19 @@
+﻿using KramDeliveryFood_v3.Enums;
+using System;
+using System.Collections.Generic;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Order
+    {
+        public Guid OrderId { get; set; }
+        public List<Product> OrderProducts = new();
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:10]: { get; set; }

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Provider.cs [hunk 13]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace KramDeliveryFood_v3.Models
+{
+    public class Provider
+    {
+        public Guid ProdviderId { get; set; }
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Provider.cs:7]: ProviderId

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Data/StoreContext.cs (hunk 8 above)
Comment (yevhenbilushko) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Data/StoreContext.cs:6]: please add Customer entity

File: OnlineMarket/KramDeliverFoodCompleted/Interfaces/IData.cs [hunk 14]
```diff
@@ -0,0 +1,15 @@
+﻿using KramDeliverFoodCompleted.Models;
+using System.Collections.Generic;
+
+namespace KramDeliverFoodCompleted.Interfaces
+{
+    public interface IData
```
Comment (yevhenbilushko) [inline OnlineMarket/KramDeliverFoodCompleted/Interfaces/IData.cs:6]: no need it

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs [hunk 15]
```diff
@@ -7,13 +7,14 @@ namespace KramDeliveryFood_v3.Models
...
         public Guid OrderId { get; set; }
-        public List<Product> OrderProducts = new();
+        public Guid BuyerId { get; set; }
         public Buyer Buyer { get; set; }
         public string PhoneNumber { get; set; }
         public string Address { get; set; }
         public DateTime OrderTime { get; set; }
         public decimal FullPrice { get; set; }
-        public Deliver Delivers { get; set; }
+        public Deliver Deliver { get; set; }
```
Comment (yevhenbilushko) [inline KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:16]: add DeliverId

**Review by yevhenbilushko — APPROVED** [review]

**Comment by yevhenbilushko** [comment]
u can merge it

File: KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs (hunk 15 above)
Comment (gecmenow) [reply KramDeliveryFood_v3/KramDeliveryFood_v3/Models/Order.cs:16]: Deliver is an enum, so Id will pass to nowhere i guess, how to be?

**Pull request merged**